### PR TITLE
chore(edit-page-2): Enhance performance in contentlet tools #28331

### LINKED ISSUE #28331: Edit Page V2: Enhance performance in Contentlet Tools
### Parent Issue

https://github.com/dotCMS/core/issues/27546

### Task

Transform `getXYZPosition` functions to optimize DOM Rendening in [Contentlet Tools](https://github.com/dotCMS/core/blob/959776837dd1208e661cf0a1bab5ef43ee0a50c1/core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/ema-contentlet-tools/ema-contentlet-tools.component.ts)

### Proposed Objective

Technical User Experience

### Proposed Priority

Priority 3 - Average

### Acceptance Criteria

* We should refactor all functions that are triggering in the `HTML`
* Contentlet Tools should work as expected after the refactor

### Assumptions & Initiation Needs

* You understand the impact of triggering functions inside angular templates

### PR DESCRIPTION
**Pull request by zJaaal**
### Proposed Changes
* Refactor to make function call inside an ngOnChanges instead of the DOM
* Code cleaning

### CODE REVIEW

**Review by KevinDavilaDotCMS — APPROVED** [review]

File: core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/ema-contentlet-tools/ema-contentlet-tools.component.ts [hunk 1]
```diff
@@ -85,21 +83,27 @@ export class EmaContentletToolsComponent implements OnInit, OnChanges {
...
             return;
         }
 
-        if (this.hasContentletOrContainerChanged(changes.contentletArea)) {
-            this.hideMenus();
-        }
+        this.hideMenus(); // We need to hide the menu if the contentlet changes
+        this.setVtlFiles(); // Set the VTL files for the component
+
+        this.ACTIONS_CONTAINER_WIDTH = this.contentletArea.payload.vtlFiles ? 178 : 128; // Set the width of the actions container
```
Comment (oidacra) [inline core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/ema-contentlet-tools/ema-contentlet-tools.component.ts:98]: const for the numbers.

File: core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/ema-contentlet-tools/ema-contentlet-tools.component.ts [hunk 2]
```diff
@@ -85,21 +83,27 @@ export class EmaContentletToolsComponent implements OnInit, OnChanges {
...
+
+        // If the contentlet changed, we need to update the styles
+        this.styles = {
+            bounds: this.getBoundsPosition(),
```
Comment (oidacra) [inline core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/ema-contentlet-tools/ema-contentlet-tools.component.ts:102]: This is not the same as using the method direct in the template?

Comment (zJaaal) [reply core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/ema-contentlet-tools/ema-contentlet-tools.component.ts:102]: No, in this `ngOnChanges` im sure that we don't trigger this functions unless we change the `contentletArea` input

**Review by rjvelazco — APPROVED** [review]

**Review by fmontes — APPROVED** [review]

**Comment by rjvelazco** [comment]
### Passed Internal QA

- Tested on Docker Image: [`dotcms/trunk_5921e6a`]

### Video

https://github.com/dotCMS/core/assets/72418962/823c32b9-4673-4808-82bb-b1e1d69c8808

**Pull request merged**